otx/cli/utils/multi_gpu.py

```diff
@@ -237,8 +238,23 @@ def run_child_process(
 
         MultiGPUManager.initialize_multigpu_train(rdzv_endpoint, rank, local_rank, gpu_ids, world_size)
 
+        threading.Thread(target=MultiGPUManager.check_parent_processes_alive, daemon=True).start()
+
         train_func()
 
+    @staticmethod
+    def check_parent_processes_alive():
+        """Check parent process is alive and if not, exit by itself."""
+        cur_process = psutil.Process()
+        parent = cur_process.parent()
+        while True:
```

Comment:
I have a little concern about this. Is there no need to add a timeout?

Reply:
Yes. this function is always started as daemon. So, when parent process exits, then this process also exits automatically.